sol/main.c

```diff
@@ -601,37 +666,13 @@ alloc_sol_conf(void)
 
 int
 gk_solicitor_enqueue_bulk(struct sol_config *sol_conf,
-	struct rte_mbuf **pkts, uint8_t *priorities, uint16_t num_pkts)
+	struct rte_mbuf **pkts, uint16_t num_pkts)
 {
-	int i;
-	unsigned int num_enqueued;
-	struct priority_req *req_nodes[num_pkts];
-
-	if (unlikely(rte_mempool_get_bulk(sol_conf->mb.pool,
-			(void **)req_nodes, num_pkts) != 0))
-		return -1;
-
-	for (i = 0; i < num_pkts; i++) {
-		if (priorities[i] > GK_MAX_REQ_PRIORITY) {
-			SOL_LOG(WARNING, "Trying to enqueue a request with priority %hhu, but should be in range [0, %d]. Overwrite the priority to PRIORITY_REQ_MIN (%hhu)\n",
-				priorities[i], GK_MAX_REQ_PRIORITY,
-				PRIORITY_REQ_MIN);
-			priorities[i] = PRIORITY_REQ_MIN;
-		}
```

Comment:
This test is still important to guarantee the correctness of the SOL block, and can be relocated to `enqueue_req()`.